### PR TITLE
CreateDoc View

### PR DESCRIPTION
**Pull request by Stefafrontani**
First mockup for Create Doc page

image

### CODE REVIEW

File: src/components/Nav/nav.css [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  display: flex;
+  font-size: 24px;
+  box-shadow: 0px -5px 14px 4px rgba(0, 0, 0, 0.2);
+}
+
+.nav__item {
+  padding: 30px 40px;
+}
+
+.nav__item:nth-child(2) {
```
Comment (globantPipo) [inline src/components/Nav/nav.css:11]: Is this to get the separation between logo and the nav right side links ?

File: src/components/Nav/index.tsx [hunk 2]
```diff
@@ -1,20 +1,34 @@
...
-            <li>
-                <Link to="/about">About</Link>
-            </li>
-            <li>
-                <Link to="/crear-documento">Create Doc</Link>
-            </li>
+  return (
+    <header>
+      <nav>
+        <ul className="nav">
```
Comment (globantPipo) [inline src/components/Nav/index.tsx:9]: Feels odd this list to be a single list, knowing that are separated elements. Is there any reason to join them, rather than make separated elements ?

File: src/containers/CreateDoc/createDoc.css [hunk 3]
```diff
@@ -0,0 +1,53 @@
+.page__content {
+  width: 100%;
+  max-width: 1024px;
+  margin: 0 auto;
+  display: grid;
+  grid-template: auto auto 1fr/ 30% 70%;
```
Comment (globantPipo) [inline src/containers/CreateDoc/createDoc.css:6]: NOT RELATED: Never used that before, seems usefull, but not sure if I understand it

File: src/containers/CreateDoc/createDoc.css [hunk 4]
```diff
@@ -0,0 +1,53 @@
+.page__content {
```
Comment (globantPipo) [inline src/containers/CreateDoc/createDoc.css:1]: just to double check>  is  `page` your block over here ? and `content` an element inside that block  ?

File: src/containers/CreateDoc/createDoc.css [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  grid-column: 1 / 3;
+  display: flex;
+  justify-content: space-around;
+}
+.page__content ul li {
+  padding: 20px 30px;
+}
+
+/* INFORMATION - Left aside */
+.page__content__information {
```
Comment (globantPipo) [inline src/containers/CreateDoc/createDoc.css:21]: how's BEM  working here ? looks like `page` block could be avoided (or used in really general cases) over here you want `content` to be your block and `information` to be your element.

`content__information` is an option

I've seen other approaches like `layout` block, making this to be like 
`layout__information-bar` 
Keep in mind that you can use multi word with `-` in BEM

File: src/containers/CreateDoc/index.tsx [hunk 6]
```diff
@@ -1,26 +1,120 @@
...
+        <ul className="Dont-know-what-is-this">
+          <li>Tutela</li>
+          <li>Contrato</li>
+          <li>Demanda</li>
+          <li>Sugerencia documento</li>
+        </ul>
+        <div className="page__content__information">
+          <div className="information">
+            <h4 className="information__title">¿Qué es una tutela?</h4>
+            <p className="information__explanation">
```
Comment (globantPipo) [inline src/containers/CreateDoc/index.tsx:40]: Please add a `TODO` here that all of that text info should come from a config or endpoint data.

File: src/containers/CreateDoc/index.tsx [hunk 7]
```diff
@@ -1,26 +1,120 @@
...
-import Dropdown from '../../components/Dropdown';
-import documentTypes from './config';
-import { createPdf } from './pdf';
+import React, { useState } from "react";
+import Dropdown from "../../components/Dropdown";
+import documentTypes from "./config";
+import { createPdf } from "./pdf";
+import "./createDoc.css";
+
+import Button from "@material-ui/core/Button";
```
Comment (globantPipo) [inline src/containers/CreateDoc/index.tsx:7]: Keep imports sorted, all the external libs, like react or material, should be on top, after that import the local modules.  : )

File: src/containers/CreateDoc/index.tsx [hunk 8]
```diff
@@ -1,26 +1,120 @@
...
+import GetAppIcon from "@material-ui/icons/GetApp";
+import VisibilityIcon from "@material-ui/icons/Visibility";
+import createDocStyles from "./styles";
+import DeleteIcon from "@material-ui/icons/Delete";
 
 const CreateDoc: React.FC = () => {
+  const [docProcessInitialized, handleProcessInitialization] = useState<
+    Boolean
```
Comment (globantPipo) [inline src/containers/CreateDoc/index.tsx:15]: Why do we need this indentation here ?

File: src/containers/CreateDoc/index.tsx [hunk 9]
```diff
@@ -1,26 +1,120 @@
...
-            <Dropdown name="documentType" title="Document type" options={documentTypes} />
-            <button onClick={openPdf}>See preview</button>
-            <button onClick={downloadPdf}>Download PDF</button>
-        </React.Fragment>
-    );
-}
+  return (
+    <section className="page_main">
+      <div className="page__content">
+        <ul className="Dont-know-what-is-this">
```
Comment (globantPipo) [inline src/containers/CreateDoc/index.tsx:31]: Do you still need the `Dont-know-what-is-this` class ?

File: src/containers/CreateDoc/index.tsx [hunk 10]
```diff
@@ -1,26 +1,120 @@
...
+              <Dropdown
+                name="documentType"
+                title="Document type"
+                options={documentTypes}
+              />
+              <Button
+                onClick={openPdf}
+                variant="contained"
+                color="default"
+                classes={classes}
```
Comment (globantPipo) [inline src/containers/CreateDoc/index.tsx:99]: If I'm not wrong, this `classes` are being used just for this button, can be renamed to `buttonClasses` ?

File: src/containers/CreateDoc/index.tsx [hunk 11]
```diff
@@ -1,26 +1,120 @@
...
+              Una vez inicializado el proceso, puedes cancelarlo
+              en cualquier momento haciendo click en el botón
+              "Cancelar Documento"
+            </p>
+          </div>
+        </div>
+        <div className="page__content__document">
+          {docProcessInitialized ? (
+            <Button
+              onClick={() => handleProcessInitialization(false)}
```
Comment (globantPipo) [inline src/containers/CreateDoc/index.tsx:75]: I THINK (not sure) we should keep this function name standard, like `docStatus` and `setDocStatus` I mean, usually `handle` functions are defined in the file, but this one was created with a hook. What do you think ?

File: src/containers/CreateDoc/index.tsx [hunk 12]
```diff
@@ -1,26 +1,120 @@
...
+            </Button>
+          ) : (
+              <Button
+                color="default"
+                onClick={() => handleProcessInitialization(true)}
+              >
+                Iniciar Documento
+              </Button>
+            )}
+          {docProcessInitialized ? (
```
Comment (globantPipo) [inline src/containers/CreateDoc/index.tsx:88]: There's 2 ternary operators here, is the any reason you wanted to do it this way rather than a single one ?

File: src/components/Nav/index.tsx (hunk 2 above)
Comment (Stefafrontani) [reply src/components/Nav/index.tsx:9]: They are separated in "style" but they all correspond to main app navigation, that's the reason i put it all together by the same UL. Usually the logo is used to give the user the possibility of returning to the home page (i think this is more a UX thing).
Despite this, i can tell you that the nav does not need a list inside it, quoting w3.org: `A nav element doesn't have to contain a list, it can contain other kinds of content as well.`.
So i think it would be fine to just do (pseudo code) nav hx ul li li li ul nav. It's up to you.

File: src/components/Nav/nav.css (hunk 1 above)
Comment (Stefafrontani) [reply src/components/Nav/nav.css:11]: Yes sir

File: src/containers/CreateDoc/createDoc.css (hunk 3 above)
Comment (Stefafrontani) [reply src/containers/CreateDoc/createDoc.css:6]: Its one of the multiple ways of creating a layout with the grid property.
You might know that while flexbox is used for one-dimensional layouts, the grid is very usefull when another axis comes to play (x-y)
Specifically, the grid-template is defining the composition of that template, separating the rows and the columns specs with a '/' (rows Specs / columns Specs)
Each specs used the same kind of values: rows and columns are managed the same way.
Auto: The row will took the height its content has
1fr   : Assume we want a 3 rows template and every row (or column) has 1fr as value (1fr 1fr 1r), every row would take `1/(1+1+1)` of the total height the tempalte end up occupying
%%: Each % value will tell that column to occupy that width in %

Summing up: in the example we have a grid template with 3 rows, where the first 2 will have the height of its content and two columns with the specific widths (30 % and 70 %)

File: src/containers/CreateDoc/createDoc.css (hunk 5 above)
Comment (Stefafrontani) [reply src/containers/CreateDoc/createDoc.css:21]: BEM is always a mess in my head. So different ways to use it.
In this case, you are right, information is an element inside my content block. I mantained the "page" word to know the source of that content: Its the content of the page, as a global content. Otherwise, i thought content__information might be a little bit misleading (if read with no context at all - what information? Of what content?). Had i overthought it too much? please let me know and i change it :D :D

File: src/containers/CreateDoc/index.tsx (hunk 8 above)
Comment (Stefafrontani) [reply src/containers/CreateDoc/index.tsx:15]: No need at all. Was using prettier till i got rid of it 🗡️ .
Premonition: A well-known remove unused code(indentation-ish) is coming

File: src/containers/CreateDoc/index.tsx (hunk 9 above)
Comment (Stefafrontani) [reply src/containers/CreateDoc/index.tsx:31]: Same premonition. Same same. My bad

File: src/containers/CreateDoc/index.tsx (hunk 10 above)
Comment (Stefafrontani) [reply src/containers/CreateDoc/index.tsx:99]: @ferochoteco , SOS.  What would be the best practice here?

File: src/components/Nav/index.tsx (hunk 2 above)
Comment (globantPipo) [reply src/components/Nav/index.tsx:9]: I would keep the logo inside the nav, but OUTSIDE the UL, in a separated element, so de UL is actually what is in the right side of the bar, and the logo is separated. That way you keep clean the inbetween of the elements, and also, don't need that nth-child.

In this case nth child is not that good just because is easily broken (just adding another element next to the logo)

File: src/containers/CreateDoc/createDoc.css (hunk 3 above)
Comment (globantPipo) [reply src/containers/CreateDoc/createDoc.css:6]: that sounds really complex, but usefull, I'll play aroudn with it, thanks for the explaination : )

File: src/containers/CreateDoc/createDoc.css (hunk 5 above)
Comment (globantPipo) [reply src/containers/CreateDoc/createDoc.css:21]: The main need over here is not to add multiple blocks to the same name. So define your block (it could be page or content) and then your element (information-module or whatever you feel comfortable with. 

That's it : ) 

remember: `blocks-name__elements-name--modifier`

File: src/containers/CreateDoc/index.tsx (hunk 8 above)
Comment (globantPipo) [reply src/containers/CreateDoc/index.tsx:15]: haha

File: src/containers/CreateDoc/styles.ts [hunk 13]
```diff
@@ -0,0 +1,10 @@
+import { makeStyles } from "@material-ui/core/styles";
+import { Theme } from "@material-ui/core";
+
+const createDocStyles = makeStyles((theme: Theme) => ({
+    root: {
+        margin: theme.spacing(2),
```
Comment (ferochoteco) [inline src/containers/CreateDoc/styles.ts:6]: This class is not being used?

File: src/containers/CreateDoc/index.tsx (hunk 7 above)
Comment (globantPipo) [reply src/containers/CreateDoc/index.tsx:7]: you kindamissed this comment, not a big deal but still FYI

File: src/containers/CreateDoc/index.tsx [hunk 14]
```diff
@@ -1,26 +1,121 @@
...
+import "./createDoc.css";
+
+import Button from "@material-ui/core/Button";
+import GetAppIcon from "@material-ui/icons/GetApp";
+import VisibilityIcon from "@material-ui/icons/Visibility";
+import createDocClasses from "./styles";
+import DeleteIcon from "@material-ui/icons/Delete";
 
 const CreateDoc: React.FC = () => {
+  const [docInitialized, setDocInitialization] = useState<Boolean>(false);
```
Comment (globantPipo) [inline src/containers/CreateDoc/index.tsx:14]: ATP this should be updated to redux, can you add a TODO for me please ?

File: src/containers/CreateDoc/index.tsx [hunk 15]
```diff
@@ -1,26 +1,121 @@
...
+  }
+
+  function downloadPdf(): void {
+    createPdf().download();
+  }
+
+  function handleDocInitialization(event: React.MouseEvent<HTMLButtonElement>): void {
+    const button = event.currentTarget;
+    const buttonDataAttributes = button.dataset;
+    const docinitialized = buttonDataAttributes.docinitialized || 'false';
```
Comment (globantPipo) [inline src/containers/CreateDoc/index.tsx:29]: And this probably to actions
